---
title: Ex-20-5553
date: 2025-04-26
author: Your Name
cell_count: 6
score: 5
---

```python
#created at 20250403
```


```python
#Exception chaining 
```


```python
print("\n === Exception Chaining ===")
```

    
     === Exception Chaining ===



```python
def process_file_data():
    try:
        with open('nonexistent.txt', 'r') as file:
            data = file.read()
            return int(data)
    except FileNotFoundError as e:
        raise ValueError("Could Not process the data") from e

                    
```


```python
try:
    process_file_data()
except ValueError as e:
    print(f"Error:{e}")
    if e.__cause__:
        print(f"caused by: {e.__cause__}")
```

    Error:Could Not process the data
    caused by: [Errno 2] No such file or directory: 'nonexistent.txt'



```python
#Cleanup created files
for file  in ['sample.txt', 'output.txt', 'data.csv', 'data.json', 'binary.dat', 'modes.txt', 'dict_data.csv']
    try:
        os.remove(file)
    except OSError:
        pass
```


---
**Score: 5**